DataFormats/SoATemplate/test/SoAUnitTests.cc

```diff
@@ -0,0 +1,103 @@
+#include <memory>
+#include <tuple>
+
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+// clang-format off
+GENERATE_SOA_LAYOUT(SimpleLayoutTemplate,
+  SOA_COLUMN(float, x),
+  SOA_COLUMN(float, y),
+  SOA_COLUMN(float, z),
+  SOA_COLUMN(float, t))
+// clang-format on
+
+using SimpleLayout = SimpleLayoutTemplate<>;
+
+TEST_CASE("SoATemplate") {
+  const std::size_t slSize = 10;
+  const std::size_t slBufferSize = SimpleLayout::computeDataSize(slSize);
+  std::unique_ptr<std::byte, decltype(std::free) *> slBuffer{
+      reinterpret_cast<std::byte *>(aligned_alloc(SimpleLayout::alignment, slBufferSize)), std::free};
+  SimpleLayout sl{slBuffer.get(), slSize};
+  SECTION("Row wide copies, row") {
+    SimpleLayout::View slv{sl};
+    SimpleLayout::ConstView slcv{sl};
+    auto slv0 = slv[0];
+    slv0.x() = 1;
+    slv0.y() = 2;
+    slv0.z() = 3;
+    slv0.t() = 5;
+    // Fill up
+    for (SimpleLayout::View::size_type i = 1; i < slv.metadata().size(); ++i) {
+      auto slvi = slv[i];
+      slvi = slv[i - 1];
```

Comment:
This assignment was failing without the change in `SoACommon.h` (bug in the const flavor of the mutable row accessor, triggered by the fact that the operator=() parameter is const).